src/main/resources/static/assets/js/scripts.js

```diff
@@ -0,0 +1,71 @@
+<!-- Delete Item-->
+function confirm_delete_item(id) {
+    if (window.confirm('Are you sure you want to delete?')) {
+        $.ajax({
+            type : "DELETE",
+            url : "/delete/"+ id,
+            success : function() {
+                window.location.reload();
+            },
+            error : function() {
+                alert("Something didn't work well. If the error persist, please contact us!");
+            }
+        });
+    }
+}
+
+<!-- Delete Item from View page-->
+function confirm_delete_item_from_view(id) {
+    if (window.confirm('Are you sure you want to delete?')) {
+        $.ajax({
+            type : "DELETE",
+            url : "/delete/"+ id,
+            success : function() {
+                window.location=document.referrer;
+            },
+            error : function() {
+                alert("Something didn't work well. If the error persist, please contact us!");
+            }
+        });
+    }
+}
+
+<!-- Delete Category-->
+function confirm_delete_category(id) {
+    var actual = document.URL;
+    if (window.confirm('Are you sure you want to delete?' +
+        '\nWhen you delete this category, all the items who belong of this category, will be delete!')) {
+        $.ajax({
+            type : "DELETE",
+            url : "/category/delete/"+ id,
+            success : function() {
+                window.location.replace(actual);
+            },
+            error : function() {
+                alert("Something didn't work well. If the error persist, please contact us!");
+            }
+        });
+    }
+}
+
+<!-- Validation Add Category-->
+$(document).ready(function() {
+    $("#newCategory #categoryName").keyup(function(){
+        document.getElementById("errorCategory").style.display = "none";
+        document.getElementById("saveCategory").disabled = false;
+        var name = $("#newCategory #categoryName").val();
+        var categoryRequest = new XMLHttpRequest();
+        categoryRequest.open('GET', 'categories');
+        categoryRequest.onload = function() {
+            var categoryData = JSON.parse(categoryRequest.responseText);
+            for(var i in categoryData){
+                if(categoryData[i].name.toUpperCase() == name.toUpperCase()){
+                    document.getElementById("saveCategory").disabled = true;
+                    document.getElementById("errorCategory").style.display = "block";
+                    $("#newCategory #errorCategory").text("The name already exists! Please try something else!");
```

Comment:
your approach is correct, but it's a lot of code and you definitely can make it cleaner. let me explain to you how to make it clean. you need to define in the category controller the following function: "is_category_in_use", which will check only for a parameter if is in use or not (for example, you fetch the URL "categories/{id}/in-use", and you only search for the {id} in the db, and the returned json would be: "{ success => 0 }" or { exists => 1 }", that will be 5-6 lines of code and less time consuming. please tell me if you understand

Reply:
Good ideas. Thank you. Just one thing, I used **Axios Library** instead **fecth()**, because this one is supported from all the browsers.